InternalPythonModules/android/fbmessenger.py

```diff
@@ -156,8 +158,11 @@ def analyzeContacts(self, dataSource, fileManager, context):
                 self._logger.log(Level.WARNING, "Error processing query result for account", ex)
                 self._logger.log(Level.WARNING, traceback.format_exc())
             except TskCoreException as ex:
-                self._logger.log(Level.SEVERE, "Failed to add Facebook Messenger contact artifacts.", ex)
+                self._logger.log(Level.SEVERE, "Failed to add FB Messenger contact artifacts.", ex)
                 self._logger.log(Level.SEVERE, traceback.format_exc())
+            except IllegalArgumentException as ex:
```

Comment:
All of these IllegalArgumentExeceptions should not be caught and logged. This means the query is not right/there are programming errors we need to fix. The general exception firewall in Android analyzer will handle these.

Reply:
These are from sending incorrect parameters to api - like missing a required parameter.  A lot of it may be dependant on the input data in the database.

It seems rather extreme to me to shutdown the entire module if we hit one bad row.

Reply:
@raman-bt, the purpose of unchecked exceptions is to make programming errors into
catastrophic errors. We should only be throwing IllegalArgumentExceptions
from these APIs if the developer has used them incorrectly; the unchecked
exceptions should get the developer's attention during testing.

Reply:
Got it. Removed catching the IllegalArgumentExceptions.